api/runnerpool/runner_pool.go

```diff
@@ -21,6 +21,15 @@ type RunnerPool interface {
 	Shutdown(context.Context) error
 }
 
+type PKIData struct {
+	Ca   string
+	Key  string
+	Cert string
+}
+
+// SecureRunnerFactory represents a factory method for constructing runners with PKI
+type SecureRunnerFactory func(addr string, pki *PKIData) (Runner, error)
```

Comment:
While we're at it, could we rename it MTLSRunnerFactory, since certs are only used for mutual tls? A "secure" runner could be implemented in other ways.

Reply:
I've made the change.